warehouse/query-core/src/main/java/datawave/query/util/ssdeep/SSDeepHashScorer.java

```diff
@@ -0,0 +1,119 @@
+     * @param signature1
+     *            the first object to be compared.
+     * @param signature2
+     *            the second object to be compared.
+     * @return an integer between 0 and 100
+     */
+    public int apply(SSDeepHash signature1, SSDeepHash signature2) {
+        if ((null == signature1) || (null == signature2)) {
+            return -1;
+        }
+        final long chunkSize1 = signature1.getChunkSize();
+        final long chunkSize2 = signature2.getChunkSize();
+        
+        // We require the chunk size to either be equal, or for one to be twice the other. If the chunk sizes don't
+        // match then we are comparing apples to oranges. This isn't an 'error' per se. We could have two valid
+        // ssdeep hashes, but with chunk sizes so different they can't be compared.
+        if ((chunkSize1 != chunkSize2) && (chunkSize1 != (chunkSize2 * 2)) && (chunkSize2 != (chunkSize1 * 2))) {
+            if (log.isDebugEnabled()) {
+                log.debug("block sizes too different: " + chunkSize1 + " " + chunkSize2);
+            }
+            return 0;
+        }
+        
+        // There is very little information content in sequences of the same character like 'LLLLL'. Eliminate any
+        // sequences longer than MAX_REPEATED_CHARACTERS (3).
+        final String s1chunk = SSDeepHash.normalizeSSDeepChunk(signature1.getChunk(), maxRepeatedCharacters);
+        final String s1doubleChunk = SSDeepHash.normalizeSSDeepChunk(signature1.getDoubleChunk(), maxRepeatedCharacters);
+        final String s2chunk = SSDeepHash.normalizeSSDeepChunk(signature2.getChunk(), maxRepeatedCharacters);
+        final String s2doubleChunk = SSDeepHash.normalizeSSDeepChunk(signature2.getDoubleChunk(), maxRepeatedCharacters);
+        
+        // Each ssdeep has two chunks with different chunk sizes. Choose which ones to use from each hash for scoring.
+        final long score;
+        if (chunkSize1 == chunkSize2) {
+            // The ssdeep chunk sizes are equal.
+            final long score1 = scoreChunks(s1chunk, s2chunk, chunkSize1);
+            final long score2 = scoreChunks(s1doubleChunk, s2doubleChunk, chunkSize2);
+            score = Math.max(score1, score2);
+        } else if (chunkSize1 == (chunkSize2 * 2)) {
+            // The first ssdeep has twice the chunk size of the second.
+            score = scoreChunks(s1chunk, s2doubleChunk, chunkSize1);
+        } else {
+            // The second ssdeep has twice the chunk size of the first.
+            score = scoreChunks(s1doubleChunk, s2chunk, chunkSize2);
+        }
+        
+        return (int) score;
+    }
+    
+    /**
+     * This is the low level chunk scoring algorithm. It takes two chunks and scores them on a scale of 0-100 where 0 is a terrible match and 100 is a great
+     * match. The chunkSize is used to cope with very small messages.
+     */
+    private static long scoreChunks(final String s1, final String s2, final long chunkSize) {
```

Comment:
Should this be returning an int?

Reply:
Likely another refactoring artifact, will fix. Thanks for the catch.